examples/cpp/pyperf/PyPerfNativeStackTrace.cc

```diff
@@ -191,5 +222,95 @@ bool NativeStackTrace::error_occured() const {
   return error_occurred;
 }
 
+bool NativeStackTrace::is_cached(const UnwindCache &map, const uint32_t &key) {
+  try {
+      map.at(key);
+      return true;
+  }
+  catch (const std::out_of_range&) {
+      logInfo(2, "No entry for %d in the cache\n", key);
+  }
+  return false;
+}
+
+UnwindCacheEntry NativeStackTrace::cache_get(const UnwindCache &map, const uint32_t &key) {
+  const UnwindCacheEntry & entry = map.at(key);
+  return entry;
+}
+
+// cache_put adds a new entry to the unwind cache if the capacity allows
+void NativeStackTrace::cache_put(UnwindCache &mp, const uint32_t &key, const unw_cursor_t cursor, const unw_addr_space_t as, void *upt) {  
+  // Check available capacity
+  if (cache_size() > NativeStackTrace::CacheMaxSizeMB*1024*1024 - cache_single_entry_size()) { 
+    logInfo(2, "The cache usage is %.2f MB, close to reaching the max memory usage (%d MB)\n", cache_size_KB()/1024, NativeStackTrace::CacheMaxSizeMB);
```

Comment:
`2` means it's not printed by default - only if we pass `-v 2` or higher. Correct?

(these should not be printed by default, only when debugging)

Reply:
It is not printed by default (and longer story below), but it makes sense to put here logLevel=`3` to distinguish this info from others that are more of a warning nature.

Long story short, this is related to what I reported regarding my troubles getting/finding logs from PyPerfNativeStack. Increasing verbosity on granulate/gprofiler does not make those bcc/pyperf logs visible in gprofiler stdout. Increasing [setVerbosityLevel in PyPerfLoggingHelper.cc](https://github.com/Granulate/bcc/blob/master/examples/cpp/pyperf/PyPerfLoggingHelper.cc#L14)) does not help as well.

I understand no one has reported this problem before so maybe it's my lack of knowledge of how the gprofiler handles logs coming from its modules. So, I decided to check the correctness of those new logs (especially passed values - how they look like) by dumping them into a temp file. For this purpose, I modified locally  [void logInfo](https://github.com/Granulate/bcc/blob/master/examples/cpp/pyperf/PyPerfLoggingHelper.cc#L20) to open a /tmp/myfile.txt and process vfprintf to it. Below I share the code diff. 


```
 void logInfo(uint64_t logLevel, const char* fmt, ...) {
+
+  if (logLevel <= 2 ) {
+    va_list va;
+    va_start(va, fmt);
+    FILE * pFile;
+    pFile = fopen("/tmp/gprofiler_tmp/izahelperfile.txt","a");
+    if (pFile) {
+      std::vfprintf(pFile, fmt, va);
+      fclose (pFile);
+      va_end(va);
+    }
 [...]
 }
````

Reply:
It's an interesting idea to increase verbosity in PyPerf when you increase verbosity in gProfiler indeed. It's not the case. That doesn't explain why PyPerf shows no prints though.

Reply:
I applied this diff on gProfiler and now I do see log messages in the `PyPerf output:` print:
```
diff --git a/gprofiler/profilers/python_ebpf.py b/gprofiler/profilers/python_ebpf.py
index 9113f355..649f9367 100644
--- a/gprofiler/profilers/python_ebpf.py
+++ b/gprofiler/profilers/python_ebpf.py
@@ -175,6 +175,8 @@ class PythonEbpfProfiler(ProfilerBase):
             str(self._EVENTS_BUFFER_PAGES),
             "--symbols-map-size",
             str(self._SYMBOLS_MAP_SIZE),
+            "-v",
+            "9",
             # Duration is irrelevant here, we want to run continuously.
         ] + self._offset_args()
```

I'll open a ticket to add it, it's a good feature that'll help debug PyPerf in the future.

Reply:
Created https://github.com/Granulate/gprofiler/issues/516